File: src/main/java/org/funivan/lologa/tile/TileInterface.java
package org.funivan.lologa.tile;

import org.funivan.lologa.tile.Position.PositionInterface;
import org.funivan.lologa.tile.Score.ScoreInterface;

import java.awt.*;

public interface TileInterface {

    Color color();

    ScoreInterface score();

    PositionInterface position();

    boolean same(TileInterface target);

}

File: src/main/java/org/funivan/lologa/tile/TileInterface.kt
package org.funivan.lologa.tile

import org.funivan.lologa.tile.position.PositionInterface
import org.funivan.lologa.tile.Score.ScoreInterface
import java.awt.Color

interface TileInterface {

    fun color(): Color

    fun score(): ScoreInterface

    fun position(): PositionInterface

    fun same(target: TileInterface): Boolean

}
